### PR TITLE
Fix incorrect candidate bachelor degree equivalency

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

When a candidate inputs their degrees, they can chose an option called 'Another qualification equivalent to a degree'. This free text field will try to figure out if the value imputed is a bachelor degree or not.

Previously we would just check if the free text search value is in our list of Bachelor degrees by checking if the string is included in the list. We didn't match the strings exactly.

So if you have the correct keywords, like Bachelor of Medicine and Pixie hunting we would classify this degree as a bachelor degree because there is a valid bachelor degree called `Bachelor of Medicine`.

This caused the form to break as there is no valid degree for Pixie hunting in the reference data gem.

This commit checks the strings exactly to avoid this issue. Fixing this [sentry error](https://dfe-teacher-services.sentry.io/issues/5809526550/?notification_uuid=820b4a63-536c-4ff7-a6d5-c359bf2a4844&project=1765973&referrer=regression_activity-slack) 

## Changes proposed in this pull request

The `Type of bachelor degree` should only be there for valid bachelor degrees

| Before | After |
| ------ | ----- |
| Screenshot from 2024-10-21 14-45-15 | Screenshot from 2024-10-21 15-37-11 |

## Guidance to review

Go on review app and add a `Another qualification equivalent to a degree` with the keywords of a actual bachelor degree like `Bachelor of Medicine`

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[[Bug] Editing bachelor degree result in an exception](https://trello.com/c/0oASlkHt/152-bug-editing-bachelor-degree-result-in-an-exception)

**Review by elceebee — APPROVED** [review]
I've tested on the review app and things work as expected. 

I'm happy to approve as is -- might be worth considering weather we want it to match something that has just been typed lower case (bachelor of arts in architecture), but don't want to delay the bug fix.

File: spec/components/candidate_interface/degree_review_component_spec.rb [hunk 1]
```diff
@@ -87,7 +87,151 @@
...
+          key: 'Degree type',
+          value: 'Bachelor',
+          action: {
+            text: "Change #{t('application_form.degree.qualification.change_action')} for Bachelor of Arts in Architecture, Woof, University of Doge, 2008",
+            href: Rails.application.routes.url_helpers.candidate_interface_degree_edit_path(degree1, :degree_level),
+          },
+        )
+      end
+
+      context 'with case insensitive' do
```
Comment (elceebee) [inline spec/components/candidate_interface/degree_review_component_spec.rb:131]: 👍

**Review by elceebee — APPROVED** [review]
I've tested the case insensitivity and it works as i would expect it to. Nice one!

**Pull request merged**